src/components/MyRecipes.vue

```diff
@@ -1,291 +1,566 @@
+        isValid = false;
+      }
+      if (!this.cookbook_id) {
+        this.cookbookError = "The cookbook field is required";
+        isValid = false
+      }
+      return isValid
+    },
+
+    async submitButton() {
+      const validFile = this.dropdownValidation()
+      this.submitLoading = true
+      const dataSend =
+      {
+        title: this.title,
+        nationality: this.nationality,
+        ingredients: this.ingredients,
+        keywords: this.keywords,
+        draft: "false",
+        recipeDescription: this.recipeDescription,
+        imagePath: this.imagePath,
+        cookbook_id: this.cookbook_id,
+        tags: [],
+        summary: this.summary,
+      }
+
+      if (validFile === true) {
+        const postResponse = await this.$store.dispatch('post_recipe', dataSend)
+        const result = await postResponse
+        if (result && result.status === 201) {
+          alert("Recipe has been created")
+          this.toggleEditor('hide')
+          this.submitLoading = false
+          // Clear the fields
+          this.title = '';
+          this.nationality = '';
+          this.ingredients = [{ name: '', unit: '', thumbnail: '', link: '' }];
+          this.keywords = '';
+          this.recipeDescription = '';
+          this.imagePath = '';
+          this.cookbook_id = '';
+          // Reload the page
+          location.reload();
+        } else {
+          console.log("Error")
+          this.submitLoading = false
         }
+      }
+
+      if (validFile === false) {
+        alert('You have incomplete fields')
+        this.submitLoading = false
+      }
     },
-    filters: {
-        truncate: function (text, length, suffix) {
-            if (text.length > length) {
-                return text.substring(0, length) + suffix;
-            } else {
-                return text;
-            }
-        },
+
+    async saveDrafts() {
+      const dataSend =
+      {
+        title: this.title,
+        nationality: this.nationality,
+        ingredients: this.ingredients,
+        keywords: this.keywords,
+        draft: "true",
+        recipeDescription: this.recipeDescription,
+        imagePath: this.imagePath,
+        cookbook_id: this.cookbook_id,
+        summary: this.summary,
+        tags: [],
+      }
+      const postResponse = await this.$store.dispatch('post_recipe', dataSend)
+      const result = await postResponse
+      if (result && result.status === 201) {
+        alert("Recipe has been created")
+        this.toggleEditor('hide')
+        // Clear the fields
+        this.title = '';
+        this.nationality = '';
+        this.ingredients = [{ name: '', unit: '', thumbnail: '', link: '' }];
+        this.keywords = '';
+        this.recipeDescription = '';
+        this.imagePath = '';
+        this.cookbook_id = '';
+        // Reload the page
+        location.reload();
+      } else {
+        console.log("Error")
+      }
+    }
+
+  },
```

Comment:
Additionally, I see duplication between this function and the post_recipe function, how about refactoring and abstract common parts to a reusable function to keep the code cleaner and shorter